gazebo_plugins/src/gazebo_ros_camera.cpp

```diff
@@ -356,32 +368,41 @@ void GazeboRosCamera::Load(gazebo::sensors::SensorPtr _sensor, sdf::ElementPtr _
   }
 
   for (uint64_t i = 0; i < impl_->num_cameras_; ++i) {
-    // C parameters
-    auto default_cx = (static_cast<double>(width[i]) + 1.0) / 2.0;
-    auto cx = _sdf->Get<double>("cx", default_cx).first;
-
-    auto default_cy = (static_cast<double>(height[i]) + 1.0) / 2.0;
-    auto cy = _sdf->Get<double>("cy", default_cy).first;
-
+    // Camera intrinsic parameters
+    impl_->fx_.push_back(impl_->camera_[i]->ImageFocalLengthX());
+    impl_->fy_.push_back(impl_->camera_[i]->ImageFocalLengthY());
+    impl_->cx_.push_back(impl_->camera_[i]->ImageOpticalCenterX());
+    impl_->cy_.push_back(impl_->camera_[i]->ImageOpticalCenterY());
     impl_->hfov_.push_back(impl_->camera_[i]->HFOV().Radian());
 
-    double computed_focal_length =
-      (static_cast<double>(width[i])) / (2.0 * tan(impl_->hfov_[i] / 2.0));
+    if(_sdf->HasElement("cx")) {
+      RCLCPP_WARN(
+          impl_->ros_node_->get_logger(),
+          "The <cx> [%f] you have provided for camera [%s] is "
+          "deprecated and will be ignored. Consider not providing the "
```

Comment:
I don't think we should ignore the tag. We should retain the exact same behavior as before, except with a deprecation message. The deprecation message should suggest what Gazebo tag to use alternatively.

Reply:
I see. I will make the changes. Thanks for the review.